cardinal/server/server.go

```diff
@@ -1,275 +1,121 @@
-	}, nil
+	err := s.registerHandlers()
+	return s, err
 }
 
-// Initialize initializes the server. It firsts checks for a port set on the handler via options.
-// if no port is found, or a bad port was passed into the option, it falls back to an environment variable,
-// CARDINAL_PORT. If not set, it falls back to a default port of 4040.
-func (handler *Handler) Initialize() {
-	if _, err := strconv.Atoi(handler.Port); err != nil || len(handler.Port) == 0 {
-		envPort := os.Getenv("CARDINAL_PORT")
-		if _, err = strconv.Atoi(envPort); err == nil {
-			handler.Port = envPort
-		} else {
-			handler.Port = "4040"
-		}
-	}
-	handler.server = &http.Server{
-		Addr:              fmt.Sprintf(":%s", handler.Port),
-		Handler:           handler.Mux,
-		ReadHeaderTimeout: readHeaderTimeout,
-	}
+// Port returns the port the server will run on.
+func (s *Server) Port() string {
+	return s.port
 }
 
 // Serve serves the application, blocking the calling thread.
 // Call this in a new go routine to prevent blocking.
-func (handler *Handler) Serve() error {
+func (s *Server) Serve() error {
 	hostname, err := os.Hostname()
 	if err != nil {
 		return eris.Wrap(err, "error getting hostname")
 	}
-	log.Info().Msgf("serving cardinal at %s:%s", hostname, handler.Port)
-	handler.running.Store(true)
-	err = eris.Wrap(handler.server.ListenAndServe(), "error listening and serving")
-	handler.running.Store(false)
-	return err
-}
-
-func (handler *Handler) Close() error {
-	err := eris.Wrap(handler.server.Close(), "error closing server")
+	log.Info().Msgf("serving at %s:%s", hostname, s.port)
+	s.running.Store(true)
+	err = s.app.Listen(":" + s.port)
 	if err != nil {
-		return err
+		return eris.Wrap(err, "error starting Fiber app")
 	}
+	s.running.Store(false)
 	return nil
 }
 
-func (handler *Handler) Shutdown() error {
-	handler.shutdownMutex.Lock()
-	defer handler.shutdownMutex.Unlock()
-	displayLogs := false
-	if handler.running.Load() {
-		// handler.running tracks whether the server is running.
-		// for safety allow shutdown to happen whenever this method is called
-		// EVEN if running reads that it is not running.
-		// However, only display the log message if expected running state is consistent.
-		// meaning that shutdown is called on a server where running is true.
-		displayLogs = true
+// Shutdown gracefully shuts down the server without interrupting any active connections.
+func (s *Server) Shutdown() {
+	err := s.app.Shutdown()
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to shutdown server")
 	}
+}
 
-	if displayLogs {
-		log.Info().Msg("Shutting down server.")
+func (s *Server) setupSwagger() {
+	file, err := os.CreateTemp("", "")
+	if err != nil {
+		panic("could not create temp file for swaggerFile")
 	}
-	ctx := context.Background()
-	err := eris.Wrap(handler.server.Shutdown(ctx), "error shutting down http server")
+	_, err = file.Write(swaggerData)
 	if err != nil {
-		return err
+		panic("could not write swaggerFile to temp file")
```

Comment:
This setupSwagger should return an error, and this panic should be `return err`. The caller (server.New) already returns an error, so this error can be bubbled up.

Reply:
remember to eris wrap it so users will know where the error came from.

Reply:
fixed here [remove swagger file after initializing](https://github.com/Argus-Labs/world-engine/pull/544/commits/ff1f284ee9799adada3d4e03fb65c6b40bef84c3)

Reply:
get eris in there. Removal of panic removes the stack trace.

Reply:
check the commit, the setupSwagger function adds eris wrapping